# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: 67a39d84be12
# Full Hash: 67a39d84be12a0ecb22429dbeb88937a3af44450
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-09 21:46:09
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Rework QuotaManagerService::ResetStoragesForPrincipal to use an async message with an async response; r=dom-storage-reviewers,jari
#   
#   A sub actor is no longer created. Actual result is now returned as
#   an asynchronous response to an asynchronous message.
#   
# ==============================================================================

diff -r f52dd3943988 -r 67a39d84be12 dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Wed Oct 09 11:57:46 2024 +0000
+++ b/dom/quota/ActorsParent.cpp	Wed Oct 09 11:59:34 2024 +0000
@@ -5928,6 +5928,22 @@
       });
 }
 
+RefPtr<BoolPromise> QuotaManager::ShutdownStoragesForOrigin(
+    Maybe<PersistenceType> aPersistenceType,
+    const PrincipalInfo& aPrincipalInfo, Maybe<Client::Type> aClientType) {
+  AssertIsOnOwningThread();
+
+  auto shutdownOriginOp =
+      CreateShutdownOriginOp(WrapMovingNotNullUnchecked(this), aPersistenceType,
+                             aPrincipalInfo, aClientType);
+
+  RegisterNormalOriginOp(*shutdownOriginOp);
+
+  shutdownOriginOp->RunImmediately();
+
+  return shutdownOriginOp->OnResults();
+}
+
 RefPtr<BoolPromise> QuotaManager::ShutdownStorage(
     Maybe<OriginOperationCallbackOptions> aCallbackOptions,
     Maybe<OriginOperationCallbacks&> aCallbacks) {